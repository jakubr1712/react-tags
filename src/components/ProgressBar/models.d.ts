export interface IProgressBarProps {
  progress: number;
  color: string;
}
