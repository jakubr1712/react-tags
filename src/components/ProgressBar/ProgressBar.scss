.progress-bar {
  &__container {
    display: flex;
    align-items: center;
    margin-left: 10px;
  }

  &__item {
    width: 10px;
    height: 5px;
    border-radius: 5px;
    background-color: #e0e0e0; 
    margin-right: 5px;

    &--active {
      background-color: green;
    }
  }
}
