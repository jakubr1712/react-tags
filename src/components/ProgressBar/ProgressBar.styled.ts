import styled from "styled-components";

export const ProgressBarContainer = styled.div`
  flex: 1;
  height: 4px;
  background-color: #ccc;
  border-radius: 2px;
  margin-left: 8px;
  position: relative;
`;

export const ProgressBarFill = styled.div`
  height: 100%;
  border-radius: 2px;
  width: 0;
`;
