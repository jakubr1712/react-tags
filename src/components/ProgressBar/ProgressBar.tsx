import React from "react";
import "./ProgressBar.scss";
import { IProgressBarProps } from "./models.d";

const ProgressBar: React.FC<IProgressBarProps> = ({ progress, color }) => {
  const totalDots = 5;
  const activeDots = Math.round((progress / 100) * totalDots);

  const barsAll = Array.from({ length: totalDots }).map((_, index) => (
    <div
      key={index}
      role={
        index < activeDots ? "progressbar-item--active" : "progressbar-item"
      }
      className={`progress-bar__item ${
        index < activeDots ? "progress-bar__item--active" : ""
      }`}
      style={index < activeDots ? { backgroundColor: color } : {}}
    />
  ));

  return <div className="progress-bar__container">{barsAll}</div>;
};

export default ProgressBar;
