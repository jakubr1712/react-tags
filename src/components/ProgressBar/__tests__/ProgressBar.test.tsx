import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProgressBar from "../ProgressBar";

describe("ProgressBar component", () => {
  it("renders properly and matches snapshot", () => {
    const { container } = render(<ProgressBar progress={0} color="#000" />);
    expect(container).toMatchSnapshot();
  });

  it("renders the correct number of dots", () => {
    render(<ProgressBar progress={0} color="#000" />);
    expect(screen.getAllByRole("progressbar-item")).toHaveLength(5);
  });

  it("renders the correct number of active dots", () => {
    render(<ProgressBar progress={60} color="#000" />);
    const activeDots = screen.getAllByRole("progressbar-item--active");
    expect(activeDots).toHaveLength(3);
  });

  it("renders with correct color for active dots", () => {
    const activeColor = "#4caf50";
    render(<ProgressBar progress={60} color={activeColor} />);
    const activeDots = screen.getAllByRole("progressbar-item--active");
    activeDots.forEach((dot) => {
      expect(dot).toHaveStyle(`background-color: ${activeColor}`);
    });
  });
});
