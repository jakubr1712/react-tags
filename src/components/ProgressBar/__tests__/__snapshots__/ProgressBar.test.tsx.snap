// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component renders properly and matches snapshot 1`] = `
<div>
  <div
    class="progress-bar__container"
  >
    <div
      class="progress-bar__item "
      role="progressbar-item"
    />
    <div
      class="progress-bar__item "
      role="progressbar-item"
    />
    <div
      class="progress-bar__item "
      role="progressbar-item"
    />
    <div
      class="progress-bar__item "
      role="progressbar-item"
    />
    <div
      class="progress-bar__item "
      role="progressbar-item"
    />
  </div>
</div>
`;
